File: VB/CoordinateSystems/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraMap

Namespace CoordinateSystems

    Public Partial Class Form1
        Inherits Form

        Const geoMapGeoDataFile As String = "../../Data/Countries.shp"

        Const cartesianMapCartesianDataFile As String = "../../Data/Hotel1.shp"

        Const geoMapCartesianDataShpFile As String = "../../Data/Sweden_county07.shp"

        Const geoMapCartesianDataPrjFile As String = "../../Data/Sweden_county07.prj"

        Private Property Adapter As ShapefileDataAdapter

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Overloads Sub OnLoad(ByVal sender As Object, ByVal e As EventArgs)
            Adapter = New ShapefileDataAdapter()
            vectorLayer.Data = Adapter
            AddHandler vectorLayer.DataLoaded, AddressOf DataLayer_DataLoaded
        End Sub

'#Region "#GeoMapGeoData"
        Private Sub SetGeoMapGeoData()
            imageLayer.Visible = False
            Dim baseUri As Uri = New Uri(Reflection.Assembly.GetEntryAssembly().Location)
            Adapter.FileUri = New Uri(baseUri, geoMapGeoDataFile)
            Adapter.SourceCoordinateSystem = New GeoSourceCoordinateSystem()
            mapControl.CoordinateSystem = New GeoMapCoordinateSystem()
        End Sub

'#End Region  ' #GeoMapGeoData
'#Region "#CartesianMapCartesianData"
        Private Sub SetCartesianMapCartesianData()
            imageLayer.Visible = False
            Dim baseUri As Uri = New Uri(Reflection.Assembly.GetEntryAssembly().Location)
            Adapter.FileUri = New Uri(baseUri, cartesianMapCartesianDataFile)
            Adapter.SourceCoordinateSystem = New CartesianSourceCoordinateSystem()
            mapControl.CoordinateSystem = New CartesianMapCoordinateSystem()
        End Sub

'#End Region  ' #CartesianMapCartesianData
'#Region "#GeoMapCartesianData"
        Private Sub SetGeoMapCartesianData()
            imageLayer.Visible = True
            Dim baseUri As Uri = New Uri(Reflection.Assembly.GetEntryAssembly().Location)
            Adapter.FileUri = New Uri(baseUri, geoMapCartesianDataShpFile)
            Adapter.SourceCoordinateSystem = ShapefileDataAdapter.LoadPrjFile(New Uri(baseUri, geoMapCartesianDataPrjFile))
            mapControl.CoordinateSystem = New GeoMapCoordinateSystem()
        End Sub

'#End Region  ' #GeoMapCartesianData
        Private Sub DataLayer_DataLoaded(ByVal sender As Object, ByVal e As DataLoadedEventArgs)
            mapControl.ZoomToFitLayerItems(0.4)
        End Sub

        Private Sub cbDataType_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs)
            Select Case cbDataType.SelectedIndex
                Case 0
                    SetGeoMapGeoData()
                Case 1
                    SetCartesianMapCartesianData()
                Case 2
                    SetGeoMapCartesianData()
                Case Else
                    SetGeoMapGeoData()
            End Select
        End Sub
    End Class
End Namespace

File: CS/CoordinateSystems/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraMap;

namespace CoordinateSystems {
    public partial class Form1 : Form {
        const string geoMapGeoDataFile = "../../Data/Countries.shp";
        const string cartesianMapCartesianDataFile = "../../Data/Hotel1.shp";
        const string geoMapCartesianDataShpFile = "../../Data/Sweden_county07.shp";
        const string geoMapCartesianDataPrjFile = "../../Data/Sweden_county07.prj";

        ShapefileDataAdapter Adapter { get; set; }

        public Form1() {
            InitializeComponent();
        }

        private void OnLoad(object sender, EventArgs e) {
            Adapter = new ShapefileDataAdapter();
            vectorLayer.Data = Adapter;
            vectorLayer.DataLoaded += DataLayer_DataLoaded;
        }

        #region #GeoMapGeoData
        void SetGeoMapGeoData() {
            imageLayer.Visible = false;

            Uri baseUri = new Uri(System.Reflection.Assembly.GetEntryAssembly().Location);
            Adapter.FileUri = new Uri(baseUri, geoMapGeoDataFile);
            Adapter.SourceCoordinateSystem = new GeoSourceCoordinateSystem();

            mapControl.CoordinateSystem = new GeoMapCoordinateSystem();
        }
        #endregion #GeoMapGeoData

        #region #CartesianMapCartesianData
        void SetCartesianMapCartesianData() {
            imageLayer.Visible = false;

            Uri baseUri = new Uri(System.Reflection.Assembly.GetEntryAssembly().Location);
            Adapter.FileUri = new Uri(baseUri, cartesianMapCartesianDataFile);
            Adapter.SourceCoordinateSystem = new CartesianSourceCoordinateSystem();

            mapControl.CoordinateSystem = new CartesianMapCoordinateSystem();
        }
        #endregion #CartesianMapCartesianData

        #region #GeoMapCartesianData
        void SetGeoMapCartesianData() {
            imageLayer.Visible = true;

            Uri baseUri = new Uri(System.Reflection.Assembly.GetEntryAssembly().Location);
            Adapter.FileUri = new Uri(baseUri, geoMapCartesianDataShpFile);
            Adapter.SourceCoordinateSystem = ShapefileDataAdapter.LoadPrjFile(new Uri(baseUri, geoMapCartesianDataPrjFile));
            
            mapControl.CoordinateSystem = new GeoMapCoordinateSystem();
        }
        #endregion #GeoMapCartesianData

        void DataLayer_DataLoaded(object sender, DataLoadedEventArgs e) {
            mapControl.ZoomToFitLayerItems(0.4);
        }

        private void cbDataType_SelectedIndexChanged(object sender, EventArgs e) {
            switch (cbDataType.SelectedIndex) {
                case 0:
                    SetGeoMapGeoData();
                    break;
                case 1:
                    SetCartesianMapCartesianData();
                    break;
                case 2:
                    SetGeoMapCartesianData();
                    break;
                default:
                    SetGeoMapGeoData();
                    break;
            }
        }
    }
}
